src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.netcoreapp.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Security.Cryptography.Rsa.Tests
+{
+    public sealed class EncryptDecrypt_Span : EncryptDecrypt
+    {
+        protected override byte[] Encrypt(RSA rsa, byte[] data, RSAEncryptionPadding padding) =>
+            TryWithOutputArray(dest => rsa.TryEncrypt(data, dest, padding, out int bytesWritten) ? (true, bytesWritten) : (false, 0));
+
+        protected override byte[] Decrypt(RSA rsa, byte[] data, RSAEncryptionPadding padding) =>
+            TryWithOutputArray(dest => rsa.TryDecrypt(data, dest, padding, out int bytesWritten) ? (true, bytesWritten) : (false, 0));
+
+        private static byte[] TryWithOutputArray(Func<byte[], (bool, int)> func)
+        {
+            for (int length = 256; ; length = checked(length * 2))
+            {
+                var result = new byte[length];
+                var (success, bytesWritten) = func(result);
+                if (success)
+                {
+                    Array.Resize(ref result, bytesWritten);
+                    return result;
+                }
+            }
+        }
+
+        [Fact]
+        public void Decrypt_VariousSizeSpans_Success()
+        {
+            using (RSA rsa = RSAFactory.Create())
+            {
+                rsa.ImportParameters(TestData.RSA1024Params);
+                byte[] cipherBytes = Encrypt(rsa, TestData.HelloBytes, RSAEncryptionPadding.OaepSHA1);
+                byte[] actual;
+                int bytesWritten;
+
+                // Too small
+                actual = new byte[TestData.HelloBytes.Length - 1];
+                Assert.False(rsa.TryDecrypt(cipherBytes, actual, RSAEncryptionPadding.OaepSHA1, out bytesWritten));
+                Assert.Equal(0, bytesWritten);
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
> We should test the failure case, too 

We are, above. Seems wrong in this case to assert that it must fail, though. I could change it to just do the asserts on Windows and still make the call regardless if that's help.

> Because most of the algorithm tests should be being keyed off of capabilities for the class they're testing

Ok.